pkg/plugin/v2/api.go

```diff
@@ -131,6 +134,11 @@ func (p *createAPIPlugin) Validate() error {
 		return err
 	}
 
+	// check if main.go is present in the root directory
+	if _, err := os.Stat(DefaultMainPath); os.IsNotExist(err) {
```

Comment:
Will doing this break the current workflow that allows `kubebuilder create api` to create all necessary files even though it returns an error?

Reply:
It would be a bug fix but I agree with you. let's apply it only in v3+ plugin. @prafull01 could you please do this change?

/hold 
until add only to v2

Reply:
Thanks for validating my suspicions.

Reply:
Done.. Thanks @christopherhein for pointing it out.